### PR TITLE
Made ExpirationManager parameters configurable via system property

### PR DESCRIPTION
**Pull request by ahmetmircik**
closes https://github.com/hazelcast/hazelcast/issues/8277

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/map/impl/eviction/ExpirationManager.java [hunk 1]
```diff
@@ -219,4 +256,19 @@ private void updateLastCleanupTimesBeforeSorting(List<PartitionContainer> partit
             partitionContainer.setLastCleanupTimeCopy(partitionContainer.getLastCleanupTime());
         }
     }
+
+    // used for testing purposes
+    public int getTaskPeriodSeconds() {
```
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/map/impl/eviction/ExpirationManager.java]: does it have to `public` ?

File: hazelcast/src/test/java/com/hazelcast/map/impl/eviction/ExpirationManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+import static com.hazelcast.map.impl.MapService.SERVICE_NAME;
+import static com.hazelcast.map.impl.eviction.ExpirationManager.SYS_PROP_EXPIRATION_CLEANUP_OPERATION_COUNT;
+import static com.hazelcast.map.impl.eviction.ExpirationManager.SYS_PROP_EXPIRATION_CLEANUP_PERCENTAGE;
+import static com.hazelcast.map.impl.eviction.ExpirationManager.SYS_PROP_EXPIRATION_TASK_PERIOD_SECONDS;
+import static java.lang.String.valueOf;
+import static java.lang.System.clearProperty;
+import static java.lang.System.setProperty;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastParallelClassRunner.class)
```
Comment (jerrinot) [inline hazelcast/src/test/java/com/hazelcast/map/impl/eviction/ExpirationManagerTest.java]: I guess these tests should not be run in parallel - as they are changing global state - the property.

File: hazelcast/src/test/java/com/hazelcast/map/impl/eviction/ExpirationManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void testTaskPeriodSeconds_set_viaSystemProperty() throws Exception {
+        int expectedPeriodSeconds = 12;
+        setProperty(SYS_PROP_EXPIRATION_TASK_PERIOD_SECONDS, valueOf(expectedPeriodSeconds));
+        int actualTaskPeriodSeconds = new ExpirationManager(getMapServiceContext()).getTaskPeriodSeconds();
+        clearProperty(SYS_PROP_EXPIRATION_TASK_PERIOD_SECONDS);
```
Comment (jerrinot) [inline hazelcast/src/test/java/com/hazelcast/map/impl/eviction/ExpirationManagerTest.java]: it's safer to store the original property and restore it. just clearing might have unexpected side-effects - it assume the property was not set before the test was executed. also the cleaning should be done in a `finally` block.

**Comment by jerrinot** [comment]
minor things, otherwise +1

**Comment by eminn** [comment]
👍

File: hazelcast/src/main/java/com/hazelcast/map/impl/eviction/ExpirationManager.java (hunk 1 above)
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/map/impl/eviction/ExpirationManager.java]: fixed.

File: hazelcast/src/test/java/com/hazelcast/map/impl/eviction/ExpirationManagerTest.java (hunk 2 above)
Comment (ahmetmircik) [inline hazelcast/src/test/java/com/hazelcast/map/impl/eviction/ExpirationManagerTest.java]: fixed.

File: hazelcast/src/test/java/com/hazelcast/map/impl/eviction/ExpirationManagerTest.java (hunk 3 above)
Comment (ahmetmircik) [inline hazelcast/src/test/java/com/hazelcast/map/impl/eviction/ExpirationManagerTest.java]: fixed cleaning.

**Pull request merged**